pkg/virt-handler/vm.go

```diff
@@ -2450,22 +2464,41 @@ func (d *VirtualMachineController) vmUpdateHelperDefault(origVMI *v1.VirtualMach
 	vmi := origVMI.DeepCopy()
 	// Find preallocated volumes
 	var preallocatedVolumes []string
+	var changedDisks []string
 	for _, v := range vmi.Spec.Volumes {
 		var name string
 		source := v.VolumeSource
 		if source.PersistentVolumeClaim != nil || source.DataVolume != nil {
 			if source.PersistentVolumeClaim != nil {
 				name = source.PersistentVolumeClaim.ClaimName
-			} else {
+			} else if source.DataVolume != nil {
 				name = source.DataVolume.Name
+			} else {
+				continue
 			}
+
+			log.Log.Infof("name is %s", name)
 			pvc, err := d.clientset.CoreV1().PersistentVolumeClaims(vmi.Namespace).Get(context.Background(), name, metav1.GetOptions{})
 			if err != nil {
 				continue
 			}
+
+			log.Log.Infof("Got a PVC %v", pvc)
 			if pvcutils.IsPreallocated(pvc.ObjectMeta.Annotations) {
 				preallocatedVolumes = append(preallocatedVolumes, v.Name)
 			}
+			expectedDiskSize, ok := pvcutils.ExpectedDiskSize(d.cdiConfigInformer.GetStore(), d.scInformer.GetStore(), pvc)
+			log.Log.Infof("Expected disk size is %d", expectedDiskSize)
+			if ok && d.clusterConfig.ExpandDisksEnabled() {
+				d.diskSizesMtx.Lock()
+				currentSize, ok := d.diskSizes[v.Name]
```

Comment:
won't there be name collisions here?

Reply:
Last commit should resolve this issue. Thanks! it also fixes the migration test issue.